extensions/controllers/sandboxclaim_controller.go

```diff
@@ -341,3 +363,171 @@ func (r *SandboxClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+		}
+
+		// 3. Translate Ingress Rules
+		var ingressRules []networkingv1.NetworkPolicyIngressRule
+		templateNP := template.Spec.NetworkPolicy
+
+		// A. Ingress from Controller/IPs/Other Pods
+		var ingressPeers []networkingv1.NetworkPolicyPeer
+
+		// L7XLB Selector (Direct Assignment! No map iteration needed)
+		if sel := templateNP.IngressControllerSelectors; sel != nil {
+			peer := networkingv1.NetworkPolicyPeer{}
+			// Since APIs now match, we can just assign the pointer
+			peer.NamespaceSelector = sel.NamespaceSelector
+			peer.PodSelector = sel.PodSelector
+			ingressPeers = append(ingressPeers, peer)
+		}
+
+		// IP Blocks
+		for _, block := range templateNP.IngressFromIPBlocks {
+			ingressPeers = append(ingressPeers, networkingv1.NetworkPolicyPeer{
+				IPBlock: &networkingv1.IPBlock{CIDR: block.CIDR},
+			})
+		}
+
+		// Additional Rules (Direct Assignment!)
+		for _, rule := range templateNP.AdditionalIngressRules {
+			peer := networkingv1.NetworkPolicyPeer{}
+			peer.NamespaceSelector = rule.InNamespaceSelector
+			peer.PodSelector = rule.FromPodSelector
+			ingressPeers = append(ingressPeers, peer)
+		}
+
+		// B. Allow traffic to Container Ports
+		var ingressPorts []networkingv1.NetworkPolicyPort
+		if len(template.Spec.PodTemplate.Spec.Containers) > 0 {
+			for _, container := range template.Spec.PodTemplate.Spec.Containers {
+				for _, port := range container.Ports {
+					p := intstr.FromInt(int(port.ContainerPort))
+					proto := corev1.ProtocolTCP
+					if port.Protocol != "" {
+						proto = port.Protocol
+					}
+					ingressPorts = append(ingressPorts, networkingv1.NetworkPolicyPort{
+						Protocol: &proto,
+						Port:     &p,
+					})
+				}
+			}
+		}
+
+		// Only append ingress rule if we have peers or ports defined
+		if len(ingressPeers) > 0 || len(ingressPorts) > 0 {
+			ingressRules = append(ingressRules, networkingv1.NetworkPolicyIngressRule{
+				From:  ingressPeers,
+				Ports: ingressPorts,
+			})
+		}
+		np.Spec.Ingress = ingressRules
+
+		// 4. Translate Egress Rules
+		var egressRules []networkingv1.NetworkPolicyEgressRule
+
+		// Default DNS Rule (Port 53 Allow-All)
+		dnsPort53 := intstr.FromInt(53)
+		protoUDP := corev1.ProtocolUDP
+		protoTCP := corev1.ProtocolTCP
+		egressRules = append(egressRules, networkingv1.NetworkPolicyEgressRule{
+			Ports: []networkingv1.NetworkPolicyPort{
+				{Protocol: &protoUDP, Port: &dnsPort53},
+				{Protocol: &protoTCP, Port: &dnsPort53},
+			},
+		})
+
+		// Additional User Rules
+		for _, rule := range templateNP.AdditionalEgressRules {
+			var egressPeers []networkingv1.NetworkPolicyPeer
+			var egressPorts []networkingv1.NetworkPolicyPort
+
+			// Peer Logic
+			if rule.ToIPBlock != nil {
+				egressPeers = append(egressPeers, networkingv1.NetworkPolicyPeer{
+					IPBlock: &networkingv1.IPBlock{CIDR: rule.ToIPBlock.CIDR, Except: rule.ToIPBlock.Except},
+				})
+			} else if rule.ToPodSelector != nil || rule.InNamespaceSelector != nil {
```

Comment:
If a user specifies both ToIPBlock and ToPodSelector in a single EgressRule, the controller will silently ignore the ToPodSelector and only create the IPBlock rule.

Reply:
Ohhh yes, true i'll remove the else statement. This allows the controller to process both fields if they are both present in the same rule.